inc/media_offload.php

```diff
@@ -68,6 +68,43 @@ class Optml_Media_Offload extends Optml_App_Replacer {
 	 * @var string Used when moving the file to our servers.
 	 */
 	private static $last_deduplicated_original;
+
+	/**
+	 * Adds page meta query args
+	 *
+	 * @param string $action The action for which the args are needed.
+	 * @param array  $args The initial args without the added meta_query args.
+	 * @return array The args with the added meta_query args.
+	 */
+	public static function add_page_meta_query_args( $action, $args ) {
+		if ( $action === 'offload_images' ) {
+			$args['meta_query'] = [
+				'relation' => 'AND',
+				[
+					'key' => self::POST_OFFLOADED_FLAG,
+					'compare' => 'NOT EXISTS',
+				],
+			];
+		}
+		if ( $action === 'rollback_images' ) {
+			$args['meta_query'] = [
+				'relation' => 'AND',
+				[
+					'key' => self::POST_ROLLBACK_FLAG,
+					'compare' => 'NOT EXISTS',
+				],
+			];
+			if ( get_option( 'optimole_wp_install', 0 ) > 1666123200 ) {
```

Comment:
not super important but we can cache the option value inside a static var to avoid calling it each time. 

static $is_legacy_install = null; 
if($is_legacy_install === null ) { $is_legacy_install = get_option( 'optimole_wp_install', 0 ) > 1666123200 }